src/map_blocks.c

```diff
@@ -535,6 +541,152 @@ long delete_unwanted_things_from_liquid_slab(MapSlabCoord slb_x, MapSlabCoord sl
     return removed_num;
 }
 
+unsigned long remove_unwanted_things_from_wall_slab(MapSlabCoord slb_x, MapSlabCoord slb_y)
+{
+    SubtlCodedCoords stl_num = get_subtile_number_at_slab_center(slb_x, slb_y);
+    struct Thing *thing;
+    struct Map *mapblk;
+    unsigned long removed_num = 0;
+    unsigned long k;
+    long i;
+    long n;
+    for (n=0; n < AROUND_MAP_LENGTH; n++)
+    {
+        mapblk = get_map_block_at_pos(stl_num+around_map[n]);
+        k = 0;
+        i = get_mapwho_thing_index(mapblk);
+        while (i != 0)
+        {
+            thing = thing_get(i);
+            if (thing_is_invalid(thing))
+            {
+                WARNLOG("Jump out of things array");
+                break;
+            }
+            i = thing->next_on_mapblk;
+            // Per thing code
+            if (thing_in_wall_at(thing, &thing->mappos))
+            {
+                switch(thing->class_id)
+                {
+                    case TCls_Door:
+                    {
+                        // using destroy_door places claimed path, which we don't want
+                        remove_key_on_door(thing);
+                        delete_thing_structure(thing, 0);
+                        removed_num++;
+                        break;
+                    }
+                    case TCls_Effect:
+                    {
+                        destroy_effect_thing(thing);
+                        removed_num++;
+                        break;
+                    }
+                    case TCls_Object:
+                    {
+                        if ( (object_is_gold(thing)) || (thing_is_spellbook(thing)) || (thing_is_special_box(thing)) || (thing_is_workshop_crate(thing)) )
```

Comment:
Wouldn't it be better to destroy or move objects based on the OMF_DestroyedOnRoomPlace object property?

Reply:
This is walls. You're now hard-coding gold, books, boxes and crates, but I dislike hard coding stuff, so I figured to keep this configurable and consistent, but not worth adding it's own property for. The function `delete_room_slabbed_objects` has the code for OMF_DestroyedOnRoomPlace , which seems basic enough,... but what objects would do worse with that property?

Reply:
Barrels, potions, statues, etc, are destroyed on room placement but left intact when path is claimed. Going by OMF_DestroyedOnRoomPlace would change that.

Reply:
But this is on wall-placement right? So if you place a wall on a Barrel or Statue, you'd want it destroyed. And with your code you already do so too.

Reply:
So you want OMF_DestroyedOnRoomPlace to apply to walls too? And everything else gets moved?